File: 509. Fibonacci Number/solution.cpp
class Solution {
public:
  int fib(int n) {
    vector<int> memo;
    for (int index = 0; index < n + 1; index++){
      if (index == 0 || index == 1){
        memo.push_back(index);
      }
      else{
        memo.push_back(memo[index - 1] + memo[index - 2]);
      }
    }
    return memo[memo.size() - 1];
  }
};

File: 509. Fibonacci Number/solution.py
class Solution(object):
  def fib(self, n):
    """
    :type n: int
    :rtype: int
    """
    memo = []
    for index in range(0, n + 1):
      if index == 0 or index == 1:
        memo.append(index)
      else:
        memo.append(memo[index - 1] + memo[index - 2])
    return memo[-1]